geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/ParameterRequirements/ClusterParameterRequirements.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.redis.internal.ParameterRequirements;
+
+import static org.apache.geode.redis.internal.RedisConstants.ERROR_UNKNOWN_CLUSTER_SUBCOMMAND;
+
+import org.apache.geode.redis.internal.netty.Command;
+import org.apache.geode.redis.internal.netty.ExecutionHandlerContext;
+
+public class ClusterParameterRequirements implements ParameterRequirements {
+  @Override
+  public void checkParameters(Command command, ExecutionHandlerContext context) {
+    int numberOfArguments = command.getProcessedCommand().size();
+
+    if (numberOfArguments < 2) {
+      throw new RedisParametersMismatchException(command.wrongNumberOfArgumentsErrorMessage());
+    } else if (numberOfArguments == 2) {
+      confirmKnownSubcommands(command);
+    } else { // numberOfArguments > 3
+      throw new RedisParametersMismatchException(
+          String.format(ERROR_UNKNOWN_CLUSTER_SUBCOMMAND, command.getStringKey()));
+    }
+  }
+
+  private void confirmKnownSubcommands(Command command) {
+    if (!command.getStringKey().equalsIgnoreCase("slots") &&
```

Comment:
While this command shouldn't be frequently invoked this pattern should be avoided here and elsewhere in the code.

`String.equalsIgnoreCase` has to convert mismatching characters to uppercase to compare them. In this case we invoked this method up to 3 times. An alternative:

```java
final String subCommand = command.getStringKey().toLower();
if (!subCommand.equals("slots") && ...) {
  ...
}
```
At most converts the string to lower case 1 time. 

Extra credit for avoid the string conversation in `getStringKey` and look at the just bytes of this command component directly. 

```java
final byte[] subCommand = getProcessedCommand().get(1);
if (subCommand[0] == 's' || subCommand[0] == 'S') {
  if (subCommand[1] == 'l' || subCommand[1] == 'L') {
    ..
  }
} else if (subCommand[0] == 'I' ...) {
} else {
  throw ...;
}
```

Reply:
Removed this check in favor of checking in the `Executor`.

Reply:
Removed all of this in favor of just leaving the check in `ClusterExecutor`.